lib/BackbeatConsumer.js

```diff
@@ -542,6 +547,68 @@ class BackbeatConsumer extends EventEmitter {
         return subscriptions.length > 0;
     }
 
+    /**
+     * Helper method to wait for consumers to be assigned to partitions before
+     * proceeding to send canary messages.
+     * @param {number} wait - current accumulated wait time
+     * @param {callback} cb - callback to invoke
+     * @return {undefined}
+     */
+    _waitForAssignment(wait, cb) {
+        setTimeout(() => {
+            const assignments = this._consumer.assignments();
+            if (assignments.length === 0) {
+                if (wait > 50000) {
+                    return cb(true);
+                }
+                return this._waitForAssignment(wait + 2000, cb);
+            }
+            return cb();
+        }, 2000);
+    }
+
+    /**
+     * On start, send canary (noop) messages to set partition offsets for this
+     * consumer group. Necessary to avoid pause/resume bug where new locations
+     * on pause will not be able to queue entries as no current offset to
+     * determine number of entries (lag) to consume on resume.
+     * @param {callback} cb - callback to invoke
+     * @return {undefined}
+     */
+    _sendCanary(cb) {
+        const crrTopic = withTopicPrefix(replicationTopic);
+        if (this._topic !== crrTopic) {
+            return process.nextTick(cb);
+        }
+        return this._waitForAssignment(0, err => {
+            if (err) {
+                this._log.debug('could not send canary on consumer init', {
+                    method: 'BackbeatConsumer._sendCanary',
+                });
+                return cb();
+            }
+            const topicDetails = this._consumer._metadata.topics
+                .find(topic => topic.name === crrTopic);
+            const entries = topicDetails.partitions.map(p => {
+                const contents = '{"canary":true}';
+                return {
+                    key: 'canary',
+                    message: contents,
+                    partition: p.id,
+                };
+            });
+            const producer = new BackbeatProducer({
```

Comment:
Can this be part of init step, so that you don't have to create a new producer and close it every time someone hits pause/resume?

Reply:
This is within the QueueProcessor `start` process. Pause/resume requests won't go through here unless we have a paused site on startup. If queueprocessor goes down and is being brought up again, it'll send the canary again even though its not a new location.
Pause/resume requests will call the kafka consumer subscribe/unsubscribe directly